### PR TITLE
Max pods per node

### PR DESCRIPTION
**Pull request by ingvagabund**
Building on top of https://github.com/kubernetes-incubator/descheduler/pull/69

### CODE REVIEW

**Comment by aveshagarwal** [comment]
Can one of the admins verify this patch?

**Comment by ingvagabund** [comment]
@ravisantoshgudimetla @aveshagarwal PTAL
Let's do the Victor the `Evictor` kinda refactoring in another PR.

File: Makefile [hunk 1]
```diff
@@ -46,3 +46,9 @@ test-unit:
 
 test-e2e:
 	./test/run-e2e-tests.sh
```
Comment (ravisantoshgudimetla) [inline Makefile]: Thanks for adding this

File: cmd/descheduler/app/options/options.go [hunk 2]
```diff
@@ -55,4 +55,6 @@ func (rs *DeschedulerServer) AddFlags(fs *pflag.FlagSet) {
 	fs.BoolVar(&rs.DryRun, "dry-run", rs.DryRun, "execute descheduler in dry run mode.")
 	// node-selector query causes descheduler to run only on nodes that matches the node labels in the query
 	fs.StringVar(&rs.NodeSelector, "node-selector", rs.NodeSelector, "Selector (label query) to filter on, supports '=', '==', and '!='.(e.g. -l key1=value1,key2=value2)")
+	// max-no-pods-to-evict limits the maximum number of pods to be evicted per node by descheduler.
+	fs.IntVar(&rs.MaxNoOfPodsToEvictPerNode, "max-no-pods-to-evict", rs.MaxNoOfPodsToEvictPerNode, "Limits the maximum number of pods to be evicted per node by descheduler")
```
Comment (aveshagarwal) [inline cmd/descheduler/app/options/options.go]: what about MaxPodsToEvictPerNode?

**Comment by aveshagarwal** [comment]
I just glanced and in general looks good to me.

File: cmd/descheduler/app/options/options.go (hunk 2 above)
Comment (ingvagabund) [reply cmd/descheduler/app/options/options.go]: Given it is a variable name and not the flag/option name, we can change it later to make the name more clear. Though, `MaxNoOfPodsToEvictPerNode` ~ `MaxPodsToEvictPerNode`.

Comment (aveshagarwal) [reply cmd/descheduler/app/options/options.go]: That's fine. But now i think to make the flag : max-pods-to-evict-per-node to make it clear in the flag itself that its per node

Comment (ingvagabund) [reply cmd/descheduler/app/options/options.go]: Done

**Pull request merged**